packages/@aws-cdk/aws-iotevents/lib/state.ts

```diff
@@ -123,33 +140,43 @@ export class State {
     return this.props.onEnter?.some(event => event.condition) ?? false;
   }
 
-  private toStateJson(): CfnDetectorModel.StateProperty {
+  private toStateJson(scope: Construct, actionBindOptions: ActionBindOptions): CfnDetectorModel.StateProperty {
     const { onEnter } = this.props;
     return {
       stateName: this.stateName,
-      onEnter: onEnter && { events: toEventsJson(onEnter) },
+      onEnter: onEnter && { events: toEventsJson(scope, actionBindOptions, onEnter) },
       onInput: {
-        transitionEvents: toTransitionEventsJson(this.transitionEvents),
+        transitionEvents: toTransitionEventsJson(scope, actionBindOptions, this.transitionEvents),
       },
     };
   }
 }
 
-function toEventsJson(events: Event[]): CfnDetectorModel.EventProperty[] {
+function toEventsJson(
+  scope: Construct,
+  actionBindOptions: ActionBindOptions,
+  events: Event[],
+): CfnDetectorModel.EventProperty[] | undefined {
```

Comment:
When does this return `undefined`?

Reply:
AH.. It is a remnant of big refactoring..